### DISCUSSION TITLE
execute a script after disconnect

### DISCUSSION THREAD

**Discussion by guedz45** [drakkan/sftpgo#582] [General]
Hello,

thanks for the work done / for sftpgo. It is really a good very useful software. 
in order to deploy it for my use, i will need to detect sftp session disconnection to run a script. 
as far as I have identified how to launch a script at the connection as much I have not found a method for the disconnection.
Is there a way to do it? 
if no maybe an idea for new features ?

Thanks for your help
brest regards

**Comment by drakkan** [comment]
Hi,

this is currently not supported, I'll try to add this feature before v2.2.0. A such hook will work with SFTP/FTP. WeìbDAV and HTTP client will not be supported since HTTP is a stateless protocol

**Comment by guedz45** [comment]
Hi,

thanks

**Reply by drakkan** [reply]
Hi,

SSH is a multiplexing protocol, a client can open a single SSH connection and then open multiple channels to execute SSH commands or SFTP connections (each SFTP session or SSH command is executed over a separate channel of the same connection) or it can disconnect without doing anything.

I suppose you are actually using the post connect hook. The post connect hook is executed when the SSH connection is established, do you expect to have a disconnection event for each SSH connection or for each channel (SFTP, SSH command) opened over a single SSH connections or both? Can you please better explain your use case? Thanks

**Reply by drakkan** [reply]
This is now supported: https://github.com/drakkan/sftpgo/commit/570964deb382edc747a874b1683535defcc0b85c

**Comment by guedz45** [comment]
Hi drakkan,
big thanks for your help.
I wait the deb package to test your work.
Best regards

**Reply by drakkan** [reply]
Hi,

distro packages are automatically built after each commit, you can find the updated deb here

https://github.com/drakkan/sftpgo/actions/runs/1400048986

if you want an update via PPA you have to wait for v2.2.0

**Comment by guedz** [comment]
Hello,

thanks again.
according to my first tests it seems to answer my request perfectly.
